### PR TITLE
[SHIP-2813] Align FinalityTagEnabled chain config between CCIP and Core 

### PR DESCRIPTION
**Pull request by Madalosso**
[Align FinaltyTagEnabled across all products SHIP-2813](https://smartcontract-it.atlassian.net/browse/SHIP-2813)

This PR carries CCIP chain default configurations for `FinalityTagEnabled` and applies them to core configs.

### CODE REVIEW

**Comment by Madalosso** [comment]
I'll trigger soak tests on three random networks to test this

**Comment by dimriou** [comment]
You need to run `make config-docs` to update CONFIG.md file

File: core/chains/evm/config/toml/defaults/Arbitrum_Sepolia.toml [hunk 1]
```diff
@@ -3,6 +3,7 @@ ChainType = 'arbitrum'
 NoNewHeadsThreshold = '0'
 OCR.ContractConfirmations = 1
 LogPollInterval = '1s'
+FinalityTagEnabled = true
```
Comment (amit-momin) [inline core/chains/evm/config/toml/defaults/Arbitrum_Sepolia.toml]: nit: just to keep the ordering consistent across all of the configs, could you move `FinalityTagEnabled` under `ChainType` here?
```suggestion
FinalityTagEnabled = true
NoNewHeadsThreshold = '0'
OCR.ContractConfirmations = 1
LogPollInterval = '1s'
```

File: core/chains/evm/config/toml/defaults/BSC_Mainnet.toml [hunk 2]
```diff
@@ -3,6 +3,7 @@
 # There are 21 BSC validators so theoretically finality should occur after 21/2+1 = 11 blocks
 ChainID = '56'
 LinkContractAddress = '0x404460C6A5EdE2D891e8297795264fDe62ADBB75'
+FinalityTagEnabled = true
```
Comment (amit-momin) [inline core/chains/evm/config/toml/defaults/BSC_Mainnet.toml]: nit: same here
```suggestion
ChainID = '56'
FinalityTagEnabled = true
LinkContractAddress = '0x404460C6A5EdE2D891e8297795264fDe62ADBB75'
```

File: core/chains/evm/config/toml/defaults/Base_Sepolia.toml [hunk 3]
```diff
@@ -1,4 +1,5 @@
 ChainID = '84532'
+FinalityTagEnabled = true
 ChainType = 'optimismBedrock'
```
Comment (amit-momin) [inline core/chains/evm/config/toml/defaults/Base_Sepolia.toml]:
```suggestion
ChainType = 'optimismBedrock'
FinalityTagEnabled = true
```

**Comment by Madalosso** [comment]
@amit-momin thanks for your input, I've reordered the configs to match how they are displayed on the docs.

**Review by amit-momin — APPROVED** [review]

**Comment by Madalosso** [comment]
@vyzaldysanchez can we get your review on this?

**Review by vyzaldysanchez — APPROVED** [review]

**Comment by Madalosso** [comment]
I'm reflecting on this task while trying to have a better understanding of the repo, and I'm curious about one thing:
This task aligned the `FinalityTagEnabled` chain config between the CCIP repo and the default chain config on the core.

Now, this repo also contains a `ccip/config/evm` directory, which already carried the `FinalityTagEnabled` chain value from CCIP repo to the respective chains. 

1. Am I correct to assume these ccip/config/evm/<chain>.toml files are overriding the core configs? 

    a. If so, wouldn't it make sense to remove the entries from `ccip/config/evm` if there are no differences between the file content against the equivalent config file under `core/chains/evm/config/toml/default`?

2.  Or maybe ccip/config/evm configs will always be necessary, regardless of core/.../config/toml/default files?


On another topic: How does the CCIP repo gets updated with the latest core repo changes?

**Comment by friedemannf** [comment]
One thing to keep in mind is that https://github.com/smartcontractkit/ccip isn't just a package, it is a fork of https://github.com/smartcontractkit/chainlink. You backported downstream changes to upstream, from ccip => core. 

1. As ccip is a fork, it contains the same code + the ccip implementation. The ccip repository builds its own separate binary based on the code in that repository, the defaults found there don't just overwrite the core defaults, they *are* the defaults when building ccip.
2. We cannot remove any defaults from core, as this is used for everything but ccip.

CCIP is regularly updated to receive the upstream changes, with a slight delay. E.g. CCIP is currently on `v2.14.0` while chainlink `v2.16.0` is already released. Those changes made in 2.15 and 2.16 will be ported to ccip soon.

**Pull request merged**